billing-api/billing/__init__.py

```diff
@@ -68,10 +70,21 @@ def get_projects(client):
 @app.route('/reports', methods=['GET'])
 @authenticate
 def calculate_cost_by_user(client):
-    original_start_date = parse(request.args.get('fromDate'), ignoretz=True)
-    original_end_date = parse(request.args.get('toDate'), ignoretz=True)
     projects = request.args.get('projects')
     bucket_size = request.args.get('bucket')
+
+    try:
+        if 'fromDate' in request.args:
+            original_start_date = parse(request.args.get('fromDate'), ignoretz=True)
+        else:
+            original_start_date = datetime(year=datetime.today().year, month=1, day=1)
+        if 'toDate' in request.args:
+            original_end_date = parse(request.args.get('toDate'), ignoretz=True)
+        else:
+            original_end_date = datetime.today()
+    except ValueError:
+        raise BadRequestError('Please define fromDate and toDate in the format YYYY-MM-DD')
```

Comment:
Anyway you can define a helper like `checkRequest(args, fields, msg)` that you can reuse for the different endpoints?